src/main/java/teammates/ui/webapi/action/GetCourseAction.java

```diff
@@ -26,9 +28,10 @@ public void checkSpecificAccessControl() {
     @Override
     public ActionResult execute() {
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-
         CourseAttributes courseAttributes = logic.getCourse(courseId);
-        return new JsonResult(new CourseInfo.CourseResponse(courseAttributes));
+        if (courseAttributes == null) {
+            throw new EntityNotFoundException(new EntityDoesNotExistException("No course with id: " + courseId));
```

Comment:
Perhaps we should stick to the convention of returning a 404 `JsonResult` here? Thats what I have been doing in my other PRs.